### PR TITLE
Measure copy in matMul

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR adds a new function `alpaka::test::integ::measureRunTimeMs` and then uses this function to measure and print the copy times in the matMul integration test.

### CODE REVIEW

File: test/integ/matMul/src/matMul.cpp [hunk 1]
```diff
@@ -228,10 +228,16 @@ TEMPLATE_LIST_TEST_CASE("matMul", "[matMul]", TestAccs)
...
-    alpaka::wait(queueHost);
-    alpaka::memcpy(queueAcc, bufCAcc, bufCHost);
+    std::cout << "Input copy time:  "
+              << alpaka::test::integ::measureRunTimeMs(
+                     [&]
+                     {
+                         alpaka::memcpy(queueAcc, bufAAcc, bufAHost);
+                         alpaka::memcpy(queueAcc, bufBAcc, bufBHost);
+                         alpaka::memcpy(queueAcc, bufCAcc, bufCHost);
+                         alpaka::wait(queueHost);
```
Comment (bernhardmgruber) [inline test/integ/matMul/src/matMul.cpp]: I moved the wait to the last command to have the total copy time of all 3 buffers.

Comment (bernhardmgruber) [reply test/integ/matMul/src/matMul.cpp]: Ok, that was a bug :D The `wait` ensures that the `memset` on `bufCHost` finished before copying from it.

**Comment by bernhardmgruber** [comment]
Given the jittering of the copy measurement (see #1597), I wonder if there are better ways in alpaka for measuring the time of an `alpaka::memcpy`?

**Comment by bussmann** [comment]
@j-stephan should have a library for this...

**Comment by bernhardmgruber** [comment]
> @j-stephan should have a library for this...

I think a fully fledged instrumentation library might be an overkill for taking a measurement of three buffer transfers in an integration test :S But I would love to hear about how @j-stephan would measure the transfer time.

**Comment by bussmann** [comment]
That was my point 😉

**Comment by j-stephan** [comment]
This is basically a trace of single alpaka call. Using bactria for this use case is certainly possible but introduces some programming overhead since it isn't really intended for something this simple.

In the future (I'm aiming for alpaka 0.10 here) we will hopefully have bactria integrated into alpaka which will enable us to automatically gather performance data for each alpaka call. We'd need to discuss on how to integrate this into the CI, though. Do we want to print everything to `stdout`? Do we want to save a `*.csv` file somewhere? Do we want to write into a database? Do we want to do some weird blockchain stuff like @psychocoderHPC once suggested? And someone would need to write a bactria output plugin for this.

**Review by j-stephan — APPROVED** [review]

**Pull request merged**